src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -10797,29 +10811,43 @@ def _rolling_agg(
             )
 
         # Throw NotImplementedError if any parameter is unsupported
-        check_is_rolling_window_supported_by_snowflake(rolling_kwargs)
+        if window_func == "rolling":
+            check_is_rolling_window_supported_by_snowflake(window_kwargs)
+        else:
+            check_is_expanding_window_supported_by_snowflake(window_kwargs)
+
         frame = query_compiler._modin_frame.ensure_row_position_column()
         row_position_quoted_identifier = frame.row_position_snowflake_quoted_identifier
         if center:
             # -(window // 2) is equivalent to window // 2 PRECEDING
             rows_between_start = -(window // 2)  # type: ignore
             rows_between_end = (window - 1) // 2  # type: ignore
         else:
-            # 1 - window is equivalent to window - 1 PRECEDING
-            rows_between_start = 1 - window  # type: ignore
+            if window_func == "rolling":
+                # 1 - window is equivalent to window - 1 PRECEDING
+                rows_between_start = 1 - window  # type: ignore
+            else:
+                rows_between_start = Window.UNBOUNDED_PRECEDING
             rows_between_end = Window.CURRENT_ROW
 
         window_expr = Window.orderBy(col(row_position_quoted_identifier)).rows_between(
             rows_between_start, rows_between_end
         )
 
+        # Handle case where min_periods = None
+        min_periods = 0 if min_periods is None else min_periods
         # Perform Aggregation over the window_expr
         new_frame = frame.update_snowflake_quoted_identifiers_with_expressions(
             {
                 quoted_identifier: iff(
-                    count(col(quoted_identifier)).over(window_expr) >= min_periods,
+                    count(col(row_position_quoted_identifier)).over(window_expr)
+                    >= min_periods
+                    if agg_func == "count"
```

Comment:
can you  add some explanation here about why we need those special case here?

Reply:
Added.